File: src/test/java/com/uwetrottmann/trakt5/services/GenresTest.java
/*
 * Copyright 2024 Uwe Trottmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.uwetrottmann.trakt5.services;

import com.uwetrottmann.trakt5.BaseTestCase;
import com.uwetrottmann.trakt5.entities.Genre;
import org.junit.Test;

import java.io.IOException;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class GenresTest extends BaseTestCase {

    @Test
    public void test_genres_shows() throws IOException {
        List<Genre> genres = executeCall(getTrakt().genres().shows());
        assertGenres(genres);
    }

    @Test
    public void test_genres_movies() throws IOException {
        List<Genre> genres = executeCall(getTrakt().genres().movies());
        assertGenres(genres);
    }

    private void assertGenres(List<Genre> genres) {
        for (Genre genre : genres) {
            assertThat(genre.name).isNotEmpty();
            assertThat(genre.slug).isNotEmpty();
        }
    }

}

File: src/test/kotlin/com/redissi/trakt/services/GenresTest.kt
package com.redissi.trakt.services

import com.redissi.trakt.BaseTestCase
import com.redissi.trakt.entities.Genre
import kotlinx.coroutines.runBlocking
import org.amshove.kluent.shouldNotBeNull
import org.amshove.kluent.shouldNotBeNullOrEmpty
import org.junit.Test
import java.io.IOException

class GenresTest : BaseTestCase() {
    @Test
    @Throws(IOException::class)
    fun `get genres for shows`() = runBlocking {
        val genres = trakt.genres().shows()
        genres.shouldNotBeNull()
        assertGenres(genres)
    }

    @Test
    @Throws(IOException::class)
    fun `get genres for movies`() = runBlocking {
        val genres = trakt.genres().movies()
        genres.shouldNotBeNull()
        assertGenres(genres)
    }

    private fun assertGenres(genres: List<Genre>) {
        for (genre in genres) {
            genre.name.shouldNotBeNullOrEmpty()
            genre.slug.shouldNotBeNullOrEmpty()
        }
    }
}